GSoC26_H/src/baseline/gemma_zero_shot.py

```
@@ -0,0 +1,363 @@
+            sentence=sentence,
+            triples=triples,
+            raw_output=raw_output or raw,
+            parse_error=None if triples else f"No valid triples parsed from: {raw[:200]}"
+        )
+
+    def _parse_json_safe(self, text: str) -> dict:
+        """Extract JSON from model output, handling common formatting issues."""
+        text = text.strip()
+
+        # Remove markdown code fences if present
+        text = re.sub(r"```(?:json)?", "", text).strip().rstrip("```").strip()
```

Ruff (0.15.14)

[warning] Using `.strip()` with multi-character strings is misleading

(B005)